packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -694,7 +792,62 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     return linkMarks;
   }
 
+  private findTextBlocksInRange(
+    node: ProsemirrorNode,
+    range: {
+      from: number;
+      to: number;
+    },
+  ): Array<{ text: string; positionStart: number }> {
+    const nodesWithPos: Array<{
+      node: ProsemirrorNode;
+      pos: number;
+    }> = [];
+    const predicate: (node: ProsemirrorNode) => boolean = (node) =>
+      !node.isTextblock || !node.type.allowsMarkType(this.type);
+
+    // define a placeholder for leaf nodes to calculate link position
+    node.nodesBetween(range.from, range.to, (node, pos) => {
+      if (!predicate(node)) {
+        nodesWithPos.push({
+          node,
+          pos,
+        });
+      }
+    });
+
+    return nodesWithPos.map((textBlock) => ({
+      text: node.textBetween(
+        textBlock.pos,
+        textBlock.pos + textBlock.node.nodeSize,
+        undefined,
+        ' ',
+      ),
+      positionStart: textBlock.pos,
+    }));
+  }
+
+  private addLinkProperties({
+    from,
+    to,
+    href,
+    ...link
+  }: FoundAutoLink & { from: number; to: number }): Omit<FoundAutoLink, 'href'> & {
```

Comment:
Can we create a named interfaces here, rather than using `&`?

Reply:
Yes let me clean this up